codegen/smithy-kotlin-codegen/src/test/kotlin/software/amazon/smithy/kotlin/codegen/rendering/EnumGeneratorTest.kt

```diff
@@ -153,18 +152,94 @@ public sealed class Baz {
         /**
          * Convert a raw value to one of the sealed variants or [SdkUnknown]
          */
-        public fun fromValue(str: kotlin.String): test.model.Baz = when(str) {
+        public fun fromValue(v: kotlin.String): test.model.Baz = when (v) {
             "t2.micro" -> T2Micro
             "t2.nano" -> T2Nano
-            else -> SdkUnknown(str)
+            else -> SdkUnknown(v)
+        }
+
+        /**
+         * Get a list of all possible variants
+         */
+        public fun values(): kotlin.collections.List<test.model.Baz> = listOf(
+            T2Micro,
+            T2Nano,
+        )
+    }
+}
+"""
+        contents.shouldContainOnlyOnceWithDiff(expectedEnumDecl)
+    }
+
+    @Test
+    fun `it generates int enums`() {
+        val model = """
+            @documentation("Documentation for this enum")
+            intEnum Baz {
+                T2_NANO = 2
+
+                X9_OMEGA = 9999
+
+                @documentation("Documentation for this value")
+                T2_MICRO = 1
+            }
+        """.prependNamespaceAndService(version = "2", namespace = "test")
+            .toSmithyModel()
+
+        val provider = KotlinCodegenPlugin.createSymbolProvider(model, rootNamespace = "test")
+        val shape = model.expectShape<IntEnumShape>("test#Baz")
+        val symbol = provider.toSymbol(shape)
+        val writer = KotlinWriter(TestModelDefault.NAMESPACE)
+        EnumGenerator(shape, symbol, writer).render()
+        val contents = writer.toString()
+
+        val expectedEnumDecl = """
+/**
+ * Documentation for this enum
+ */
+public sealed class Baz {
+    public abstract val value: kotlin.Int
+
+    /**
+     * Documentation for this value
+     */
+    public object T2Micro : test.model.Baz() {
+        override val value: kotlin.Int = 1
+        override fun toString(): kotlin.String = value.toString()
+    }
+
+    public object T2Nano : test.model.Baz() {
+        override val value: kotlin.Int = 2
+        override fun toString(): kotlin.String = value.toString()
```

Comment:
**Question**: Is this the `toString()` representation we want for this? I know we use the value for string enums as the `toString()` representation. I suppose that is also questionable (although I think we did that for serialization purposes but I don't recall for sure at this point).

Reply:
Probably not. It mattered much less for strings, since the name and value were (generally) the same.

Seems perhaps we should commonize to `EnumSymbol.VariantSymbol`?  e.g.
```kt
override fun toString(): kotlin.String = "Baz.T2Nano"
```

Note that for serialization (both string and int enums) we use the variant value directly.

Reply:
Strictly the variant symbol (and _not_ the enum symbol plus `.`) is most consistent with how Kotlin enums are handled.